### DISCUSSION TITLE
Class file has wrong version 61.0, should be 52.0

### DISCUSSION THREAD

**Discussion by nswdxqq** [jOOQ/jOOQ#14207] [Q&A] [answered]
I'm using java 8 with latest version (3.17.5) for jOOQ (open source edition) from maven repository, got error "Class file has wrong version 61.0, should be 52.0", does jOOQ still support java 8 now?

**Comment by lukaseder** [comment, marked as answer]
Thanks a lot for your message.

The last version of the jOOQ Open Source Edition that supported Java 8 was jOOQ 3.14. You can see what Java versions are supported by what editions on the downloads page:
https://www.jooq.org/download/versions

In short:

- jOOQ 3.14: Java 8
- jOOQ 3.15: Java 11
- jOOQ 3.16: Java 11
- jOOQ 3.17: Java 17

Our commercial editions still have Java 8 distributions.

**Reply by nswdxqq** [reply]
OK, thanks!
